### PR TITLE
Refactor allies controller logic to a service

### PR DESCRIPTION
**Pull request by rhiroyuki**
Moved some private controller methods from `AlliesController` to services.

### CODE REVIEW

**Comment by rhiroyuki** [comment]
Partially solves  #840

**Comment by camillevilla** [comment]
Thanks for contributing , @rhiroyuki! I'll have a review to you by June 11th.

File: app/services/allyship_creator.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    remove_request
+    notifier.perform(pusher_type: pusher_type,
+                     current_user: current_user,
+                     ally_id: ally_id)
+  end
+
+  private
+
+  def allyship
+    return @allyship if defined?(@allyship)
```
Comment (julianguyen) [inline app/services/allyship_creator.rb]: Why not do `@allyship if @allyship`? (I haven't written Ruby in like 3 months lol)

**Comment by camillevilla** [comment]
Thanks for #905, @rhiroyuki! I just merged those changes into your branch. And now we wait...

File: app/services/allyship_creator.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    @allyship = Allyship.find_by(ally_id: ally_id, user_id: current_user.id)
+  end
+
+  def valid?
+    ally_id || current_user
```
Comment (camillevilla) [inline app/services/allyship_creator.rb]: @rhiroyuki Can you help me understand? I thought `AllyshipCreator` requires both `ally_id` and `current_user`. Should this be `&&`?

File: spec/controllers/allies_controller_spec.rb [hunk 3]
```diff
@@ -74,10 +79,7 @@
         end
 
         it 'deletes the allyship request notification' do
-          allow(Notification).to receive(:where).and_return(notification)
-          allow(notification).to receive(:order).and_return(notification)
-          expect(notification).to receive(:destroy_all)
-          subject
+          expect { subject }.to change { user.notifications.count }.from(1).to(0)
```
Comment (camillevilla) [inline spec/controllers/allies_controller_spec.rb]: 👏 Nice! I like this refactor. This test is shorter more agnostic to how the method is implemented.

File: app/services/allyship_creator.rb (hunk 2 above)
Comment (rhiroyuki) [reply app/services/allyship_creator.rb]: Ty... I've updated and created tests for both situations.

**Pull request merged**